lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_producer.rb

```diff
@@ -0,0 +1,445 @@
+# frozen_string_literal: true
+
+module Reporting
+  module Reports
+    module EnterpriseFeeSummary
+      class EnterpriseFeesWithTaxReportByProducer < ReportTemplate
+        attr_accessor :permissions
+
+        def initialize(user, params = {}, render: false)
+          super(user, params, render:)
+          @permissions = Permissions.new(user)
+        end
+
+        def search
+          report_line_items.orders
+        end
+
+        def order_permissions
+          @order_permissions ||= ::Permissions::Order.new(user, ransack_params)
+        end
+
+        def report_line_items
+          @report_line_items ||= Reporting::LineItems.new(order_permissions, params)
+        end
+
+        def supplier_ids_filter(line_item)
+          return true if params[:supplier_id_in].blank?
+
+          params[:supplier_id_in].include?(line_item.supplier.id.to_s)
+        end
+
+        def enterprise_fee_filtered_ids
+          return @enterprise_fee_filtered_ids unless @enterprise_fee_filtered_ids.nil?
+
+          @enterprise_fee_filtered_ids = EnterpriseFee.where(nil)
```

Comment:
This is confusing naming. I would have used a local variable like `enterprise_fees` first and then assigned to the instance variable when calling `pluck(:id)` later.